test/core/gprpp/capture_test.cc

```diff
@@ -0,0 +1,38 @@
+// Copyright 2021 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/core/lib/gprpp/capture.h"
+#include <gtest/gtest.h>
+
+namespace grpc_core {
+
+TEST(CaptureTest, Capture) {
+  auto f = Capture([](int* p) { EXPECT_EQ(*p, 42); }, 42);
+  f();
+}
+
+TEST(CaptureTest, WithArgsAndReturn) {
+  int captured = 1;
+  auto f =
+      Capture([captured](int* p, int arg) { return (captured + *p) * arg; }, 2);
```

Comment:
This doesn't compile on Visual Studio with anything less than C++17... (I ran into this before myself)

(the tests we have aren't specifying this switch for msvc, which then defaults to "latest", and I think this is around C++20 for VS2019)
 
https://godbolt.org/z/GqEv446qz

`error C2955: 'Capture': use of class template requires template argument list (x64 msvc v19.latest)`

I'm not totally sure if we should care, but maybe that's problematic still.

(works with C++17 being selected: https://godbolt.org/z/KrEvjd574)

Reply:
https://godbolt.org/z/4x8cGrq45

wrapper function was missed on godbolt